Config refactoring, few issues

astroseger opened the pull request:
This pull request fixes: #7 #78 #110 #119? #121  #124  #131 #132  

This pull request is mostly refactoring and major simplification of the code without breaking the old functionality. The main change is refactoring of the configuration, so I will describe only it.

The only one "breaking" change is the configuration of snet-cli for local enviroment. The old dirty hack will not work anymore,  but instead we introduce new simple and transparent functionality to do it (however if you want you could use the old trick, but you need to add default_gas_price in local network manually).

### Configuration file
In the new configuration ideology the current session is simply the union between session_identity and session_network.
* default_wallet_index became part of identity configuration
* default_gas_price, current_registry_at, current_multipartyescrow_at and current_singularitynettoken_at became part of network configuration
* we introduce "network" field in identity configuration. This parameter bind identity to specific network. You will not be able to use this identity with other networks (see #7 ). This parameter is optional for all identity types instead of "rpc" identity. Rpc identity should be always bind to specific network (actually to specific eth_rpc_endpoint).

It means that you can have different default_wallet_index for different identities and different default_gas_price (and current_registy_at) for different networks.

For example you can have the following configuration file:
```
[network.kovan]
default_eth_rpc_endpoint = https://kovan.infura.io
default_gas_price = 1000000000

[network.mainnet]
default_eth_rpc_endpoint = https://mainnet.infura.io
default_gas_price = 1000000000

[network.local]
default_eth_rpc_endpoint = http://localhost:8545
default_gas_price = 1000000000
current_singularitynettoken_at = 0x6e5f20669177f5bdf3703ec5ea9c4d4fe3aabd14
current_registry_at = 0x4e74fefa82e83e0964f0d9f53c68e03f7298a8b2
current_multipartyescrow_at = 0x5c7a4290f6f8ff64c69eeffdfafc8644a4ec3a4e

[ipfs]
default_ipfs_endpoint = http://localhost:5002

[session]
network = local
identity = snet-user

[identity.snet-user]
identity_type = key
private_key = 0xc71478a6d0fe44e763649de0a0deb5a080b788eefbbcf9c6f7aef0dd5dbd67e0
default_wallet_index = 5

[identity.mne0]
identity_type = mnemonic
mnemonic = test1 test2 test3
network = mainnet
default_wallet_index = 0

[identity.user2-rpc]
identity_type = rpc
network = local
default_wallet_index = 5
```
### Configuration for local environment 
Now you can do the following (I will prepare PR for update wiki).
```
# create default configuration file
snet 

# Create First identity (snet-user = first ganache)
snet identity create snet-user key --private-key 0xc71478a6d0fe44e763649de0a0deb5a080b788eefbbcf9c6f7aef0dd5dbd67e0

# switch to snet-user
snet identity snet-user

# Add local network and switch to it
snet network create local http://localhost:8545
snet network local

# switch to local  ipfs endpoint
snet set  default_ipfs_endpoint http://localhost:5002

# Configure contract addresses for local network (it will not be necessary for kovan or mainnet! )
snet set current_singularitynettoken_at 0x6e5f20669177f5bdf3703ec5ea9c4d4fe3aabd14
snet set current_registry_at            0x4e74fefa82e83e0964f0d9f53c68e03f7298a8b2
snet set current_multipartyescrow_at    0x5c7a4290f6f8ff64c69eeffdfafc8644a4ec3a4e
```

astroseger commented on snet_cli/arguments.py:
```diff
@@ -115,22 +120,27 @@ def add_identity_options(parser, config):
+    p = subparsers.add_parser("create", help="Create a new network")
+    p.set_defaults(fn="create")
+    p.add_argument("network_name", help="name of network to create")
+    p.add_argument("rpc_endpoint", help="ethereum rpc endpoint")
```
@arturgontijo
1. Concerning your first question: it was request from @ferrouswheel (#121). And I'm totally agree with him. 
2. You should use "snet identity ARTUR_ID" to actually switch to your identity. Probably it make sense to automatically switch to first created identity?

arturgontijo replied:
I agree with you too.
But If user try to check the `version` the message:
```We've created configuration file with dafault values in: /home/artur/.snet/config```
Will pop up but without version.
Maybe a better approach will be showing the message but also run the command:
```
$ snet version
We've created configuration file with dafault values in: /home/artur/.snet/config
...
snet version
    version: 0.1.7
```

2 - Yeah, I think the auto-change will be great!

astroseger commented:
According @arturgontijo review I've added the following:
1. snet-cli will automatically switch to the first created identity.
2. snet-cli will continue running after creating default config.
3. I've fixed few typos spotted by @arturgontijo and @raamb
